### PR TITLE
Japi-350 update parser

### PR DESCRIPTION
**Pull request by drealeed**
## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [x] I have created a corresponding JIRA ticket for this submission
- [x] My code follows the code style of this project.
  - [ ] I have applied the Eclipse code-format template provided.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [x] The change has been fully tested:
  - [ ] I have performed unit tests to cover my changes.
  - [ ] I have performed system test and covered possible regressions and side effects.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.

### CODE REVIEW

**Comment by drealeed** [comment]
Added unit test, plus ran the hpcc job that had failed against vault::thor::key::prod::bizlinkfull::qa::proxid::refs::l_address1 with the updated wsclient jar.

File: wsclient/src/main/java/org/hpccsystems/ws/client/antlr/EclRecord.g4 [hunk 1]
```diff
@@ -117,7 +119,7 @@ CPAREN             : ')';
 OCURLY             : '{';
 CCURLY             : '}';
 COMMA              : ',';
-SEMI               : ';';
+SEMI               : (';' | WS '=>');
```
Comment (rpastrana) [inline wsclient/src/main/java/org/hpccsystems/ws/client/antlr/EclRecord.g4:122]: This seems odd, are we treating the payload_sep equivalent to an end of line SEMI?

File: wsclient/src/test/java/org/hpccsystems/ws/client/DFUFileDetailWrapperTest.java [hunk 2]
```diff
@@ -198,6 +199,16 @@ public void testInlineWithMaxlength() throws Exception {
         assertNotNull(column);
     }
 
+    @Test
+    public void testPayload() throws Exception {
```
Comment (rpastrana) [inline wsclient/src/test/java/org/hpccsystems/ws/client/DFUFileDetailWrapperTest.java:203]: open curly should be in new line

**Review by rpastrana — COMMENTED** [review]
@drealeed looks good overall, I do have a question and a comment.

File: wsclient/src/main/java/org/hpccsystems/ws/client/antlr/EclRecord.g4 (hunk 1 above)
Comment (drealeed) [reply wsclient/src/main/java/org/hpccsystems/ws/client/antlr/EclRecord.g4:122]: It is odd, Antlr grammar is a PITA. This is the only way I was able to allow ecl field definitions that did NOT end in a ";", but ONLY when it was followed by a =>.

File: wsclient/src/test/java/org/hpccsystems/ws/client/DFUFileDetailWrapperTest.java (hunk 2 above)
Comment (drealeed) [reply wsclient/src/test/java/org/hpccsystems/ws/client/DFUFileDetailWrapperTest.java]: formatting updated

**Review by rpastrana — APPROVED** [review]
@drealeed approved

**Pull request merged**